### PR TITLE
closes #54 initial commit for mapping the player quests from FY19

### LINKED ISSUE #54: Upload Member Quests
- [ ] Upload the quests taken by Players from the 2019 data
- [ ] Create a cloud function to save player quests from 2019 upon user login

### PR DESCRIPTION
**Pull request by gerarrd**
added mapping function for playerQuest and playerPoints from FY19

### CODE REVIEW

File: src/app/core/email.service.ts [hunk 1]
```diff
@@ -25,6 +25,9 @@ export class EmailService {
     contentType: 'Text' | 'HTML' = 'Text',
     ccEmailAddress: string[] = []
   ) {
+    // Adding the default recipient
+    ccEmailAddress.push('rewards.recognitionph@infor.com');
```
Comment (nbcp) [inline src/app/core/email.service.ts]: lets not hardcode this, please add in [functions config](https://firebase.google.com/docs/functions/config-env) instead

File: functions/src/index.ts [hunk 2]
```diff
@@ -32,11 +35,67 @@ function migrateMembership(user: admin.auth.UserRecord) {
...
+        addPlayerQuest({
+          created: new Date(),
+          seasonId: data.seasonId,
+          playerId: user.uid,
+          playerName: user.displayName,
+          playerEmail: user.email,
+          teamId: teamData.teamId,
+          status: "completed",
+          submitted: data.dateSubmitted,
+          completed: data.dateCompleted,
```
Comment (nbcp) [inline functions/src/index.ts]: need to format submitted & completed dates coz these would save them as strings

File: functions/src/index.ts [hunk 3]
```diff
@@ -32,11 +35,67 @@ function migrateMembership(user: admin.auth.UserRecord) {
...
+
+      if(totalQuests > 0) {
+        addPlayerPoints(seasonId + user.uid, {
+          seasonId: seasonId,
+          playerId: user.uid,
+          playerName: user.displayName,
+          teamId: teamData.teamId,
+          totalPoints: totalPoints,
+          totalQuests: totalQuests
+        });
```
Comment (nbcp) [inline functions/src/index.ts]: please add updated date as current date

File: functions/src/index.ts [hunk 4]
```diff
@@ -16,11 +16,14 @@ export const migrateUserData = functions.region(region).auth.user().onCreate((us
...
 function migrateMembership(user: admin.auth.UserRecord) {
   return firestore.collection('teamRef').where('email', '==', user.email).get()
-    .then((values) => {
-      values.forEach((doc) => {
+    .then((values: any) => {
+      values.forEach((doc: any) => {
         const data = doc.data();
         const id = user.uid + data.teamId;
 
+        if(!(!!data.isLead))
```
Comment (nbcp) [inline functions/src/index.ts]: one ! is enough

File: src/app/core/email.service.ts (hunk 1 above)
Comment (gerarrd) [reply src/app/core/email.service.ts]: added to config

File: functions/src/index.ts (hunk 2 above)
Comment (gerarrd) [reply functions/src/index.ts]: formatted as requested

File: functions/src/index.ts (hunk 3 above)
Comment (gerarrd) [reply functions/src/index.ts]: added the updated date value using the current date

File: functions/src/index.ts (hunk 4 above)
Comment (gerarrd) [reply functions/src/index.ts]: removed unnecessary "!"

**Pull request closed without merging**